### ISSUE TITLE
Optmize and improve streams

### ISSUE THREAD

**Issue by alyapunov** [tarantool/tarantool#6316]
After #5860 several proposals were suggested. We should investigate them and maybe implement.

First of all current implementation seems to be high latency since stream's requests are passed for network to TX thread strictly one-by-one. We have two options here:
 * Move entire stream logic to TX thread. It could be also more clean and clear. But there could be performance issues.
 * Make additional request batching. This would definitely sophisticate entire solution.

Another issue that has to be discussed is current state of net_msg_max options. I personally thing that limit must be applied to total request count, but there's another opinion that it must limit only currently processing request count (excluding those requests that are in queue of streams). Perhaps both limits should be implemented.
